library/std/src/io/mod.rs

```diff
@@ -2408,6 +2445,34 @@ impl<T: BufRead, U: BufRead> BufRead for Chain<T, U> {
     fn consume(&mut self, amt: usize) {
         if !self.done_first { self.first.consume(amt) } else { self.second.consume(amt) }
     }
+
+    fn read_until(&mut self, byte: u8, buf: &mut Vec<u8>) -> Result<usize> {
+        let mut read = 0;
+        if !self.done_first {
+            read += self.first.read_until(byte, buf)?;
+
+            match buf.last() {
+                Some(b) if *b == byte => return Ok(read),
```

Comment:
If `first.read_until()` read zero bytes, then this might look at the last byte of the data that was already in `buf`.